### PR TITLE
[plugin-web-app] Move credentials from user info to http context in page steps

### LINKED ISSUE #4134: Error when trying to open the main page with userInfo
**Describe the bug**
When we're trying to open `web-application.main-page-url` which contains credential with user info using step `Given I am on main application page` we get the next error in log and Allure report:

` HTTP request for 'https://admin:admin@the-internet.herokuapp.com/basic_auth'  failed with the exception: Request URI authority contains deprecated userinfo component`

**To Reproduce**
Set the next properties:
`web-application.authentication-mode=URL` - `URL` the default value
`web.steps.page.keep-user-info-for-protocol-redirects=true`
`web-application.main-page-url=https://admin:admin@the-internet.herokuapp.com/basic_auth` - can be any with userInfo

Run step:
`Given I am on the main application page`

### PR DESCRIPTION
**Pull request by draker94**
Fixes #4134

### CODE REVIEW

File: vividus-plugin-web-app/src/main/java/org/vividus/steps/ui/web/PageSteps.java [hunk 1]
```diff
@@ -332,7 +339,12 @@ private URI updateUrlWithUserInfoForRedirects(URI pageUrl)
             try
             {
                 HttpClientContext context = HttpClientContext.create();
-                httpClient.doHttpHead(pageUrl, context);
+                URI uriToRequest = pageUrl;
+                if (uriToRequest.getUserInfo() != null)
+                {
+                    uriToRequest = moveUserInfoCredentialsToHttpContext(uriToRequest, context);
+                }
+                httpClient.doHttpHead(uriToRequest, context);
```
Comment (valfirst) [inline vividus-plugin-web-app/src/main/java/org/vividus/steps/ui/web/PageSteps.java:347]:
```suggestion
                URI pageUriToCheck = pageUrl;
                UserInfo userInfo = UriUtils.getUserInfo(pageUriToCheck);
                if (userInfo != null)
                {
                    addBasicAuthToContext(userInfo, context);
                    pageUriToCheck = UriUtils.removeUserInfo(pageUriToCheck);
                }
                httpClient.doHttpHead(pageUriToCheck, context);
```

File: vividus-plugin-web-app/src/main/java/org/vividus/steps/ui/web/PageSteps.java [hunk 2]
```diff
@@ -354,6 +366,20 @@ private URI updateUrlWithUserInfoForRedirects(URI pageUrl)
...
+        Credentials credentials = new UsernamePasswordCredentials(userInfo.getUser(),
+                userInfo.getPassword().toCharArray());
+        BasicScheme scheme = new BasicScheme();
+        scheme.initPreemptive(credentials);
+        AuthCache authCache = new BasicAuthCache();
+        HttpHost host = HttpHost.create(pageUrl);
+        authCache.put(host, scheme);
+        httpContext.setAuthCache(authCache);
+        return UriUtils.removeUserInfo(pageUrl);
+    }
```
Comment (valfirst) [inline vividus-plugin-web-app/src/main/java/org/vividus/steps/ui/web/PageSteps.java:381]:
```suggestion
    private void addBasicAuthToContext(UserInfo userInfo, HttpClientContext httpContext)
    {
        Credentials credentials = new UsernamePasswordCredentials(userInfo.getUser(),
                userInfo.getPassword().toCharArray());
        BasicScheme scheme = new BasicScheme();
        scheme.initPreemptive(credentials);
        AuthCache authCache = new BasicAuthCache();
        HttpHost host = HttpHost.create(pageUrl);
        authCache.put(host, scheme);
        httpContext.setAuthCache(authCache);
    }
```

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus-plugin-web-app/src/main/java/org/vividus/steps/ui/web/PageSteps.java (hunk 2 above)
Comment (valfirst) [reply vividus-plugin-web-app/src/main/java/org/vividus/steps/ui/web/PageSteps.java:381]: Also does it make sense to use `org.apache.hc.client5.http.ContextBuilder#preemptiveBasicAuth`?

Comment (draker94) [reply vividus-plugin-web-app/src/main/java/org/vividus/steps/ui/web/PageSteps.java]: > Also does it make sense to use `org.apache.hc.client5.http.ContextBuilder#preemptiveBasicAuth`?

Yep, this way looks pretty easier.

**Review by valfirst — APPROVED** [review]

**Review by valfirst — APPROVED** [review]

**Pull request merged**